libs/api/domains/license-service/src/lib/client/firearm-license-client/firearmLicenseMapper.ts

```diff
@@ -0,0 +1,201 @@
+    },
+    licenseInfo.collectorLicenseExpirationDate && {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Gildistími safnaraskírteinis',
+      value: formatDateString(licenseInfo.collectorLicenseExpirationDate) ?? '',
+    },
+    licenseInfo.address && {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Heimilisfang',
+      value: licenseInfo.address,
+    },
+    licenseInfo.qualifications && {
+      type: GenericLicenseDataFieldType.Group,
+      label: 'Réttindaflokkar',
+      fields: licenseInfo.qualifications.split('').map((qualification) => ({
+        type: GenericLicenseDataFieldType.Category,
+        name: qualification,
+        label: categories?.[`Flokkur ${qualification}`] ?? '',
+      })),
+    },
+    properties && {
+      type: GenericLicenseDataFieldType.Group,
+      label: 'Skotvopn í eigu leyfishafa',
+      fields: (properties.properties ?? []).map((property) => ({
+        type: GenericLicenseDataFieldType.Category,
+        fields: parseProperties(property)?.filter(
+          (Boolean as unknown) as ExcludesFalse,
+        ),
+      })),
+    },
+  ].filter((Boolean as unknown) as ExcludesFalse)
+
+  return {
+    data,
+    rawData: JSON.stringify(licenseData),
+  }
+}
+
+type ExcludesFalse = <T>(x: T | null | undefined | false | '') => x is T
+
+const parseProperties = (
+  property?: FirearmProperty,
+): Array<GenericLicenseDataField> | null => {
+  if (!property) return null
+
+  const mappedProperty = [
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Staða skotvopns',
+      value: property.category ?? '',
+    },
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Tegund',
+      value: property.typeOfFirearm ?? '',
+    },
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Heiti',
+      value: property.name ?? '',
+    },
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Númer',
+      value: property.serialNumber ?? '',
+    },
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: 'Hlaupvídd',
+      value: property.caliber ?? '',
+    },
+  ]
+  return mappedProperty
+}
+
+const parsePropertyForPkpassInput = (properties?: Array<FirearmProperty>) => {
+  if (!properties?.length) return 'Engin skráð skotvopn'
+
+  const propertyString = properties
+    .map(
+      (property) =>
+        `▶︎ ${property.typeOfFirearm} ${property.name}.\r\nHlaupvídd ${property.caliber}.\r\nNúmer ${property.serialNumber}.`,
+    )
+    .join('\r\n\r\n')
+
+  return propertyString
+}
+
+export const createPkPassDataInput = (
+  licenseInfo?: LicenseInfo | null,
+  propertyInfo?: FirearmPropertyList | null,
+  nationalId?: string,
+) => {
+  if (!licenseInfo || !nationalId) return null
+
+  const parseAddress = (address?: string) => {
+    if (!address) return
+
+    const splitArray = address.split(',')
+    return {
+      address: splitArray[0].trim(),
+      zip: splitArray[1].trim(),
```

Comment:
Runtime error if OG string does not contain `,`

Reply:
fixed